File: Leetcode/338. Counting Bits/338_Counting_Bits.java
class Solution {
    public int[] countBits(int n) {
        int result[] = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            result[i] = 1 + result[i & (i - 1)];
        }
        return result;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int n = 5;
        int result[] = solution.countBits(n);
        for (int i = 0; i < result.length; i++) {
            System.out.print(result[i] + " ");
        }
    }
}

File: Leetcode/338. Counting Bits/338_Counting_Bits.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> countBits(int n)
    {
        vector<int> dp(n + 1, 0);
        for (int i = 1; i <= n; i++)
        {
            dp[i] = dp[i >> 1] + (i & 1);
        }
        return dp;
    }
};

int main()
{
    Solution sol;
    int n = 5;
    vector<int> res = sol.countBits(n);
    for (int i = 0; i < res.size(); i++)
    {
        cout << res[i] << " ";
    }
    cout << endl;
}